crates/sui-graphql-rpc/src/zklogin_verify_signature.rs

```diff
@@ -0,0 +1,199 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::data::Db;
+use crate::error::Error;
+use crate::types::base64::Base64;
+use crate::types::chain_identifier::ChainIdentifier;
+use crate::types::dynamic_field::{DynamicField, DynamicFieldName};
+use crate::types::epoch::Epoch;
+use crate::types::sui_address::SuiAddress;
+use crate::types::type_filter::ExactTypeFilter;
+use async_graphql::*;
+use fastcrypto::encoding::Encoding;
+use fastcrypto::{encoding::Base64 as FastcryptoBase64, traits::ToFromBytes};
+use fastcrypto_zkp::bn254::zk_login_api::ZkLoginEnv;
+use im::hashmap::HashMap as ImHashMap;
+use shared_crypto::intent::{
+    AppId, Intent, IntentMessage, IntentScope, IntentVersion, PersonalMessage,
+};
+use std::str::FromStr;
+use sui_types::authenticator_state::{ActiveJwk, AuthenticatorStateInner};
+use sui_types::digests::{get_mainnet_chain_identifier, get_testnet_chain_identifier};
+use sui_types::dynamic_field::{DynamicFieldType, Field};
+use sui_types::signature::GenericSignature;
+use sui_types::signature::{AuthenticatorTrait, VerifyParams};
+use sui_types::transaction::TransactionData;
+use tracing::warn;
+
+pub(crate) struct ZkLoginVerify;
```

Comment:
thought/ suggestion: if we plan to have other functionality for zk stuff, maybe we can rename this to ZkLogin or just Zk?

Reply:
if not, then maybe we don't need this struct and we can export the `verify_zklogin_signature` fn directly

Reply:
revised